chat_test.go

```diff
@@ -50,18 +51,20 @@ func TestSendMessage(t *testing.T) {
 	chatName := "test-chat"
 	payload := []byte("test message")
 	chatMock := ChatMock{}
+	identity, err := crypto.GenerateKey()
+	require.NoError(t, err)
 
 	db, err := client.InitializeTmpDB()
 	require.NoError(t, err)
 	defer db.Close()
 
-	messenger := client.NewMessenger(&chatMock, nil, db)
-	vc := NewChatViewController(nil, nil, messenger, nil)
+	messenger := client.NewMessengerV2(identity, &chatMock, db)
+	vc := NewChatViewController(nil, nil, &messenger, nil)
 
 	err = vc.Select(client.Contact{Name: chatName, Type: client.ContactPublicRoom})
 	require.NoError(t, err)
 	// close reading loops
-	close(vc.cancel)
+	defer close(vc.cancel)
```

Comment:
because messenger doesn't know if anyone is consuming messages, and if there are no consumer then event provider will get stuck will get stuck. i wanted just to reuse ethereum Feed, but i didn't add that change to this PR